File: Python/centaur.py

# Matheus Ramos de Carvalho

def centaur(a, b = True):
    # finish
    if a < 0:
        return

    # call the upper level
    centaur(a-1, False)

    # default values
    top = '|       #'
    bottom = '\n'

    # is the first body?
    if a == 0:
        # +head
        top=' o\n/|\_____\n|       #'
    else:
        # spaces + default top
        top = (a-1)* '     '+ '! !``'+ top
    
    # is the last body?
    if b:
        # add the legs
        bottom = '\n'+(a)*'     '+'! !```! !'
    else:
        # add the back
        top += '____'

    # print centaur lvl a
    print(top + bottom, end='')


x = int(input())
centaur(x)

File: C/centaur.c
// Gabriel Penajo Machado
#include <stdio.h>

#define TRUE 1
#define FALSE 0

void centaur(int level, int finish) {
    // finish
    if (level < 0) {
        return;
    }
    // call upper level
    centaur(level-1, FALSE);

    // is head?
    if (level == 0) {
        printf(" o\n/|\\_____\n|       #");
    }
    else {
        // spaces padding
        for (int i = 1; i < level; ++i) {
            printf("     ");
        }
        printf("! !``");
        // default top value
        printf("|       #");
    }

    // is not last body (legs)?
    if (finish == FALSE) {
        // centaur's back
        printf("____");
        // default bottom value
        printf("\n");
    }
    else {
        // default bottom value
        printf("\n");
        // spaces padding
        for (int i = 0; i < level; ++i) {
            printf("     ");
        }
        // legs
        printf("! !```! !");
    }
}

int main() {
    int n;
    scanf("%d", &n);

    centaur(n, TRUE);
    return 0;
}